gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerSerializationIntegrateTest.java

```diff
@@ -0,0 +1,149 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.server;
+
+import org.apache.tinkerpop.gremlin.driver.Client;
+import org.apache.tinkerpop.gremlin.driver.Cluster;
+import org.apache.tinkerpop.gremlin.driver.remote.DriverRemoteConnection;
+import org.apache.tinkerpop.gremlin.process.traversal.AnonymousTraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Edge;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.apache.tinkerpop.gremlin.util.iterator.IteratorUtils;
+import org.apache.tinkerpop.gremlin.util.ser.AbstractMessageSerializer;
+import org.apache.tinkerpop.gremlin.util.ser.GraphBinaryMessageSerializerV1;
+import org.apache.tinkerpop.gremlin.util.ser.GraphSONMessageSerializerV2d0;
+import org.apache.tinkerpop.gremlin.util.ser.GraphSONMessageSerializerV3d0;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class GremlinServerSerializationIntegrateTest extends AbstractGremlinServerIntegrationTest {
+
+    private AbstractMessageSerializer serializer;
+    private Cluster cluster = null;
+    private Client client = null;
+    private GraphTraversalSource g = null;
+
+    public GremlinServerSerializationIntegrateTest(AbstractMessageSerializer serializer) {
+        this.serializer = serializer;
+    }
+
+    @Parameterized.Parameters
+    public static Collection serializers() {
+        return Arrays.asList(new Object[][] {
+                { new GraphBinaryMessageSerializerV1() },
+                { new GraphSONMessageSerializerV3d0() },
+                { new GraphSONMessageSerializerV2d0() }
+        });
+    }
+
+    @Before
+    public void openConnection() {
+        cluster = TestClientFactory.build().serializer(serializer).create();
+        client = cluster.connect();
+        g = AnonymousTraversalSource.traversal().withRemote(
+                DriverRemoteConnection.using(cluster, "gmodern"));
+    }
+
+    @After
+    public void closeConnection() {
+        if (cluster != null) cluster.close();
+    }
+
+    @Test
+    public void shouldDeserializeVertexPropertiesForGremlin() {
+        final Vertex vertex = client.submit("gmodern.V(1)").one().getVertex();
+
+        assertVertexWithProperties(vertex);
+    }
+
+    @Test
+    public void shouldSkipVertexPropertiesForGremlin() {
+        final Vertex vertex = client.submit("gmodern.with('materializeProperties', 'tokens').V(1)").one().getVertex();
+
+        assertEquals(Collections.emptyIterator(), vertex.properties());
```

Comment:
at risk of redundancy, but for sanity, please assert that the `T` values are present in addition to asserting the properties are not.

Reply:
added in https://github.com/apache/tinkerpop/pull/1843/commits/6884272f19b8ed97a9de3f53033e651b3fa437bf